.github/workflows/build.yml

```diff
@@ -31,141 +31,36 @@ env:
   GOPROXY: direct
 
 jobs:
+  list-variants:
+    if: github.repository == 'bottlerocket-os/bottlerocket'
+    runs-on: ubuntu-latest
+    outputs:
+      variants: ${{ steps.get-variants.outputs.variants }}
+      aarch-enemies: ${{ steps.get-variants.outputs.aarch-enemies }}
+    steps:
+      - uses: actions/checkout@v3
+      - id: get-variants
+        name: Determine variants
+        run: |
+          cd variants
+          output="variants=$(ls -d */ | cut -d'/' -f 1 | grep -v shared | grep -v target | jq -R -s -c 'split("\n")[:-1]')"
+          echo $output
+          echo $output >> $GITHUB_OUTPUT
+          output="aarch-enemies=$(ls -d */ | cut -d'/' -f 1 | grep -e "\-dev" -e "metal" -e "vmware" | jq -R -s -c 'split("\n")[:-1] | [ .[] | {"variant": ., "arch": "aarch64"}]')"
```

Comment:
nit: you can consolidate the grep with `grep -E '(^(metal|vmware)|\-dev$)'` while making sure you're only filtering on prefixes and suffixes.